### ISSUE TITLE
Attachments lost after reorganizing notes

### ISSUE THREAD

**Issue by guckuck** [laurent22/joplin#6569]
As descripted in [this posting](https://discourse.joplinapp.org/t/attachments-lost-after-reorganizing-notes/25282) Joplin lost attachments.

The problem still exists in the latest official version. I can reproduce it.

**Comment by laurent22** [comment]
> I can reproduce it.

Can you? Because that post is very long and it's not clear how it happens. If you have the exact steps on how to always replicate it, please add them in the issue here.

**Comment by guckuck** [comment]
Yes, I do. As I wrote:

I have two notes with attachments. For example:
- note 1, attachment a
- note 2, attachment b

Copy attachment b to note 1 (only the markup-link). Delete note 2. After a period of time, the markup-link is invalid, the ressource is missing.

I tried this again a few days ago.

**Comment by laurent22** [comment]
Can't replicate

**Comment by kpatzwald** [comment]
Why do you remove the label "high"? In my option, this problem should have the highest importance. Joplin is a note taking app and the biggest problem is data loss, my opinion. I lost hours for recovering dozens of very important attachments.

I know, the program is open source. You do a very impressiv job. Joplin is great!

I would like to help to find the problem. I only need some hints for a systematically search.

**Comment by guckuck** [comment]
Why do you remove the label "high"? In my option, this problem should have the highest importance. Joplin is a note taking app and the biggest problem is data loss, my opinion. I lost hours for recovering dozens of very important attachments.

I know, the program is open source. You do a very impressiv job. Joplin is great!

I would like to help to find the problem. I only need some hints for a systematically search.

**Comment by laurent22** [comment]
I need to prioritize things, and I've already spent several hours on this without being able to replicate it. I agree it's important but so are many other bugs and features.

I'm still watching the issue so if someone find out a way to replicate it, I'm happy to give it another try

**Comment by guckuck** [comment]
Yes, I know. Thanks for your words and for your work! You do a great work!

I also will try to find some usefull hints in the logs.

**Comment by laurent22** [comment]
By the way, when you copy attachments, what editor do you use (rich text or markdown?) and what operating system?

**Comment by laurent22** [comment]
I wonder if it's related to this: https://github.com/laurent22/joplin/issues/6211

**Comment by guckuck** [comment]
> By the way, when you copy attachments, what editor do you use (rich text or markdown?) and what operating system?

I use several devices and OS:
Phone: Android 11
iPad 4 with latest iPadOS
MacBook Pro, now with 12.4
Mac mini with 10.16
Surface now with Windows 11

I only use markdown.

> I wonder if it's related to this: https://github.com/laurent22/joplin/issues/6211

All my resources are relative links:
`[TITLE](:/dsfkljslfkjsdlfjdlsjljaldjlk)`

I don't think it is the same problem.

I just test it systematically again. Again, I copy a resource-link to another note and delete the original note. First, I only sync on the surface-device (Windows 11). Than I added the mobile phone (Android 11). After that, everything is fine.

Now I add the MacBook Pro with macOS 12.4. So far, no problems.

I will check (and save) the logs on all devices.
